Modules/Company/Http/Requests/CompanyGetValidationRequest.php

```diff
@@ -27,4 +28,20 @@ public function authorize() {
     public function rules() {
         return Config::get('validation.default_http_get');
     }
+
+    /*
+     * @return void
+     */
+    protected function prepareForValidation() :void {
+       if(Auth::user()->role !== 'admin') {
+           $extra['name'] = 'company_id';
+           $extra['comparison'] = 'equals';
+           $extra['attribute'] = 'company.id';
+           $extra['value'] = Auth::user()->company_id;
+           $inputs=  $this->input();
+           $currentPredicates = array_key_exists('predicates', $inputs) ? $inputs['predicates'] : $inputs['predicates'] = [];
+           array_push($currentPredicates, $extra);
+           $this->merge(['predicates' => $currentPredicates]);
```

Comment:
Ahora habiendo dicho todo eso, creo que la logica se puede simplificar un poco, como ves
` /*
     * @return void
     */
    protected function prepareForValidation() :void {
       if($this->user()->role !== 'admin') { //TODO do not use hardcoded string
           $companyPredicate = [
               'name' => "company_id",
               'comparison' => "equals",
               'attribute' => "company.id",
               'value' => $this->user()->company_id
           ];
           $currentPredicates = $this->input()['predicates'] ?? [];
           array_push($currentPredicates, $companyPredicate);
           $this->merge(['predicates' => $currentPredicates]);
        }
    }`